### PR TITLE
change placeholder to real what's new message for the ecliptic changes

### PR DESCRIPTION
**Pull request by eteq**
This PR updates the what's new entry for the changes mainly introduced in #8394 to actually have "real" content.  I think it would be good to have this in the RC so anyone testing it can see what this change is actually about. @Juanlu001 if you have a sec you might want to take a look, but @bsipocz if you want to do the RC I think it's probably best to just merge this then and if @Juanlu001 has changes we can do that between RC and release.

### CODE REVIEW

**Comment by eteq** [comment]
Oh, and we may want to try to get in a quick update to the coordinates docs with a bit of this info (separate from the what's new I mean), but I don't have time for that now, and since that's between RC and release it's ok to do a just-docs update (if not ideal...).

File: docs/whatsnew/3.2.rst [hunk 1]
```diff
@@ -82,11 +82,23 @@ from the :mod:`astropy.stats` to the :mod:`astropy.timeseries` module.
...
-===================================
-
-Placeholder. gist is that True/Mean are now treated appropriately, but this
-might change code from older versions (no good way to do deprecation)
+Additions and changes to Ecliptic Transformations
+=================================================
+
+The Ecliptic frames and associated transformations in ``astropy.coordinates``
+have been updated to correctly reflect the "True" and "Mean" terminology. In
+this release there are now ``*MeanEcliptic`` frames now include precession but
```
Comment (Juanlu001) [inline docs/whatsnew/3.2.rst]:
```suggestion
this release there are now ``*MeanEcliptic`` frames now which include precession but
```

**Comment by Juanlu001** [comment]
> Oh, and we may want to try to get in a quick update to the coordinates docs

What's the release schedule? I can try to write something on Monday morning I think.

**Comment by bsipocz** [comment]
> What's the release schedule?

I hope to get the RC out of the door tomorrow, narrative docs updates are more than welcome even after that.

**Pull request merged**